js/Heap.js

```diff
@@ -12,6 +13,60 @@ const track = (event, payload) => {
   RNHeap.track(event, flatten(payload));
 };
 
+getActiveRouteName = (navigationState) => {
+  if (!navigationState) {
+    return null;
+  }
+  const route = navigationState.routes[navigationState.index];
+  // dive into nested navigators
+  if (route.routes) {
+    return `${route.routeName}::${getActiveRouteName(route)}`;
+  }
+
+  return route.routeName;
+}
+
+let navRef = null;
+const NavigationService = {
+  setTopLevelNavigator: (ref) => {
+    navRef = ref;
+  }
+}
+
+const withHeapNavigationAutotrack = (AppContainer) => {
```

Comment:
I'd love to see this move to its own file so we can keep this one succinct (and the file containing the nav stuff separate).

Reply:
Also, since this is going to be specific to React Navigation, we should try to put that in the name of the thing somehow (as ungainly as that's going to be).  Maybe `withReactNavigationAutotrack` (I suspect we can drop the "Heap" since that will be obvious from the way it's used, i.e. Heap.withReactNavigationAutotrack)?